src/System.Private.ServiceModel/src/System/ServiceModel/Description/TypeLoader.cs

```diff
@@ -499,6 +495,56 @@ private KeyedByTypeCollection<IOperationBehavior> GetIOperationBehaviorAttribute
             }
         }
 
+        // Given a MethodInfo which could be from any type, return a MethodInfo declared by the given type 
+        // that matches in name and parameter types.  Null is returned if there is no match.
+        private static MethodInfo GetCorrespondingMethodFromType(Type type, MethodInfo methodInfo)
+        {
+            Contract.Assert(type != null);
+            Contract.Assert(methodInfo != null);
+
+            if (methodInfo.DeclaringType == type)
+            {
+                return methodInfo;
+            }
+
+            MethodInfo matchingMethod = type.GetTypeInfo().DeclaredMethods.SingleOrDefault(m => MethodsMatch(m, methodInfo));
+            return matchingMethod;
+        }
+
+        // Returns true if the given methods match in name and parameter types
+        private static bool MethodsMatch(MethodInfo method1, MethodInfo method2)
+        {
+            Contract.Assert(method1 != null);
+            Contract.Assert(method2 != null);
+
+            if (method1 == method2)
+            {
+                return true;
+            }
+
+            if (!String.Equals(method1.Name, method2.Name, StringComparison.Ordinal))
+            {
+                return false;
+            }
+
+            ParameterInfo[] parameters1 = method1.GetParameters();
+            ParameterInfo[] parameters2 = method2.GetParameters();
+            if (parameters1.Length != parameters2.Length)
+            {
+                return false;
+            }
+
+            for (int i = 0; i < parameters1.Length; ++i)
+            {
+                if (parameters1[0].ParameterType != parameters2[i].ParameterType)
+                {
+                    return false;
+                }
+            }
+
+            return true;
+        }
```

Comment:
2nd commit increases the number of items compared.  Curiously, testing MethodInfo's for being generic is not in public contract.  The changes in the 2nd commit pass the existing duplex tests on our NET Native experimental test environment.